src/coreclr/src/tools/crossgen2/ILCompiler.ReadyToRun/Compiler/DependencyAnalysis/ReadyToRunCodegenNodeFactory.cs

```diff
@@ -57,6 +58,12 @@ public sealed class NodeFactory
 
         public NameMangler NameMangler { get; }
 
+        public bool Composite { get; }
```

Comment:
`CompilationModuleGroup` might be a good place to hang these three since it already knows which modules are in compilation. I know we've tried to keep non-node state out of `NodeFactory` in the past to stop it becoming a dumping ground.

Reply:
Good point, fixed.